lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -49,6 +49,52 @@ DenseElementsAttr getConstantValue(Builder &builder, Attribute value,
   return res;
 }
 
+bool isAddPtrOffsetCombinable(Value first, Value second) {
+  auto GetConstantIntValue = [](Value val) -> std::optional<llvm::APInt> {
+    DenseElementsAttr constAttr;
+    Attribute attr;
+    auto defOp = val.getDefiningOp();
+    if (auto splatOp = llvm::dyn_cast_or_null<SplatOp>(defOp))
+      val = splatOp.getSrc();
+    else if (matchPattern(defOp, m_Constant(&constAttr)) && constAttr.isSplat())
+      attr = constAttr.getSplatValue<Attribute>();
+
+    // Check constant value.
+    llvm::APInt intVal;
+    if (matchPattern(val, m_ConstantInt(&intVal)))
+      return intVal;
+
+    // Check IntegerAttr
+    if (auto intAttr = dyn_cast_or_null<IntegerAttr>(attr))
+      return intAttr.getValue();
```

Comment:
the logic seems a bit convoluted should this part, `attr` is only set in `if(auto splatOp` why not move the code `if (auto intAttr = dyn_cast_or_null<IntegerAttr>(attr))` in there?